scripts/o3de/o3de/repo_properties.py

```diff
@@ -263,6 +264,33 @@ def _edit_objects(object_typename:str,
 
     repo_json[f'{object_typename}s_data'] = repo_objects_data
 
+def _auto_update_json(object_type: str or list,
+                      repo_path: pathlib.Path,
+                      repo_json: dict,
+                      ):
+    
+    repo_directory = os.path.dirname(repo_path)
+    expected_files = {"project.json": [],
+                      "gem.json": [],
+                      "template.json": []}
+    for directory, sub_directory, files, in os.walk(repo_directory):
+        current_depth = directory[len(repo_directory):].count(os.sep)
+        for filename in files:
+            if filename in expected_files.keys() and current_depth < 3:
```

Comment:
I think we should change this to not default to looking for gems inside gems or gems inside projects because then you have to add special logic to ignore the project gem, and also ignore gems or projects inside templates...

I think it's best to change this logic to walk the directories until we find a gem.json, project.json, template.json or there are no more subdirectories to look in.  If we find a valid .json file don't look in any more subdirectories in that path (don't look for gems, projects, templates inside the current directory)